models/weatherUpdater.js

```diff
@@ -0,0 +1,81 @@
+const Weather = require("./weatherSchema");
+const weatherApi = require("./weather-api");
+
+module.exports.updateWeather = function(weathers) {
+	return new Promise(function(resolve, reject) {
+		var savedCount = 0;
+		for (var i = 0; i < weathers.length; i++) {
+			updateWeather(weathers[i], function(err) {
+				if (err) {
+					reject(err);
+					return;
+				}
+				if (weathers.length == ++savedCount) {
+					resolve();
+				}
+			});
+		}
+	});
+}
+
+function updateWeather(dailyWeather, callback) {
```

Comment:
This function performs *save&update* at the same time. How about changing name of the function into **saveOrUpdateWeather**?